File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        longest = 0
        num_set = set(nums)
        
        for num in num_set:
            if num - 1 not in num_set:
                curr = num
                curr_streak = 1
                
                while curr + 1 in num_set:
                    curr += 1
                    curr_streak += 1
            
                longest = max(longest, curr_streak)
        
        return longest

File: turingplanet/set/128.longest-consecutive-sequence.cpp
/*
 * @lc app=leetcode id=128 lang=cpp
 *
 * [128] Longest Consecutive Sequence
 */

#include "lib/headers.h"

// @lc code=start
class Solution
{
  public:
    int longestConsecutive(vector<int> &nums)
    {
        unordered_set<int> vals(nums.begin(), nums.end());
        int                ans = 0;

        for (int i : vals)
        {
            int len = 1, cur = i;
            if (!vals.count(cur - 1))  //prevent duplicate check
            {
                while (vals.count(cur + 1))
                {
                    cur = cur + 1;
                    len++;
                }
            }
            ans = max(len, ans);
        }
        return ans;
    }
};
// @lc code=end
